File: Array/118.Pascal's_Triangle/Pascal's_Triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        if(numRows == 1) return {{1}};
        vector<vector<int>> dp = {{1}, {1, 1}};
        vector<int> tmp;
        for(int i = 2; i < numRows; ++i){
            tmp.clear();
            tmp.push_back(1);
            for(int j = 1; j < dp[i - 1].size(); ++j)
                tmp.push_back(dp[i - 1][j] + dp[i - 1][j - 1]);
            tmp.push_back(1);
            dp.push_back(tmp);
        }
        return dp;
    }
};

File: Array/118.Pascal's_Triangle/Pascal's_Triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        ans = [[1],[1,1],[1,2,1]];s = 0
        if numRows < 4 : return ans[:numRows]
        for i in range(numRows - 3):
            s = 0;curr = [1]
            for j in range(len(ans[-1])-1):
                curr.append(ans[-1][s] + ans[-1][s+1])
                s += 1
            curr.append(1)
            ans.append(curr)
        return ans